### DISCUSSION TITLE
What's the most convenient way to load all symbols data?

### DISCUSSION THREAD

**Discussion by HeidXC** [nautechsystems/nautilus_trader#1601] [Q&A] [answered]
Dear maintainers,

I tried load data for several tickers in backtest, but one of my strategy use all tickers bars in the venue, I noticed our wrangler is for one single bar type, but bar collections in my mongodb are mixed, does that mean I need to group data by their symbol first then export to csv, and then import in nautilus one by one? or is their any convenient way to import all data in my mongo all at once?

Another question is if I want to match orders, do I have to pass L1 or L2 orderbook data to nautilus so it can work? which one is preferred?

**Comment by cjdsellers** [comment, marked as answer]
Hi @VeraLyu

Yes, you'll want to group and process per instrument ID - then you can add as many as you like for backtesting (within memory limits for the low-level API, or have everything written out to parquet per the high-level API) https://docs.nautilustrader.io/nightly/tutorials/index.html.

Nautilus will use the available data to simulate the executions, so it doesn't required order book data but that would offer the most granularity. You can also just add trade ticks and it should work well.

**Reply by HeidXC** [reply]
Thanks for your reply!
